packages/react-next/src/components/MessageBar/MessageBar.base.tsx

```diff
@@ -1,210 +1,117 @@
+
+  const nativeProps = getNativeProps<React.HTMLAttributes<HTMLSpanElement>>(props, htmlElementProperties, [
+    'className',
+  ]);
+
+  const classNames: { [key in keyof IMessageBarStyles]: string } = getClassNames(styles, {
+    theme: theme!,
+    messageBarType: messageBarType || MessageBarType.info,
+    onDismiss: onDismiss !== undefined,
+    actions: actions !== undefined,
+    truncated: truncated,
+    isMultiline: isMultiline,
+    expandSingleLine: expandSingleLine,
+    className,
+  });
+
+  const expandIconProps = { iconName: expandSingleLine ? 'DoubleChevronUp' : 'DoubleChevronDown' };
+  const regionProps = actions || onDismiss ? { 'aria-describedby': labelId, role: 'region' } : {};
+  const actionsDiv = actions ? <div className={classNames.actions}>{actions}</div> : null;
+
+  const dismissButton = onDismiss ? (
+    <IconButton
+      disabled={false}
+      className={classNames.dismissal}
+      onClick={onDismiss}
+      iconProps={dismissIconProps ? dismissIconProps : { iconName: 'Clear' }}
+      title={dismissButtonAriaLabel}
+      ariaLabel={dismissButtonAriaLabel}
+    />
+  ) : null;
+
+  return (
+    <div ref={ref} className={classNames.root} {...regionProps}>
+      <div className={classNames.content}>
+        <div className={classNames.iconContainer} aria-hidden>
+          {messageBarIconProps ? (
+            <Icon {...messageBarIconProps} className={css(classNames.icon, messageBarIconProps.className)} />
+          ) : (
+            <Icon iconName={ICON_MAP[messageBarType!]} className={classNames.icon} />
+          )}
         </div>
-        {this._getActionsDiv()}
-      </div>
-    );
-  }
-
-  private _renderSingleLine(): React.ReactElement<React.HTMLAttributes<HTMLAreaElement>> {
-    return (
-      <div className={this._classNames.root} {...this._getRegionProps()}>
-        <div className={this._classNames.content}>
-          {this._getIconSpan()}
-          {this._renderInnerText()}
-          {this._getExpandSingleLine()}
-          {this._getActionsDiv()}
-          {this._getDismissSingleLine()}
+        <div className={classNames.text} id={labelId} role="status" aria-live={getAnnouncementPriority(messageBarType)}>
+          <span className={classNames.innerText} {...nativeProps}>
+            <DelayedRender>
+              <span>{children}</span>
+            </DelayedRender>
+          </span>
         </div>
+        {/* expand/collapse button */ (!isMultiline && !actionsDiv && truncated && (
+          <div className={classNames.expandSingleLine}>
+            <IconButton
+              disabled={false}
+              className={classNames.expand}
+              onClick={toggleExpandSingleLine}
+              iconProps={expandIconProps}
+              ariaLabel={overflowButtonAriaLabel}
+              aria-expanded={expandSingleLine}
+            />
+          </div>
+        )) ||
+          (onDismiss && <div className={classNames.dismissSingleLine}>{dismissButton}</div>)}
```

Comment:
I think this needs to be `onDismiss && (isMultiline ? dismissButton : <div ...>)`. Right now there are two dismiss buttons sometimes. 😆